tests/unit/test_charm.py

```diff
@@ -775,298 +790,282 @@ def test_validate_config_options(self, _charm_lib):
+        assert harness.charm.get_secret("app", "operator-password") is None
+
+        harness.set_leader(False)
+        harness.charm.set_secret("unit", "operator-password", "somesecret")
+        harness.charm.remove_secret("unit", "operator-password")
+        assert harness.charm.get_secret("unit", "operator-password") is None
+
+        harness.set_leader(True)
+        with caplog.at_level(logging.ERROR):
+            if juju_has_secrets:
+                error_message = (
+                    "Non-existing secret operator-password was attempted to be removed."
+                )
+            else:
+                error_message = (
+                    "Non-existing field 'operator-password' was attempted to be removed"
+                )
 
-            self.harness.charm.remove_secret("unit", "operator-password")
-            assert (
-                "Non-existing secret operator-password was attempted to be removed."
-                in self._caplog.text
-            )
+            harness.charm.remove_secret("app", "operator-password")
+            assert error_message in caplog.text
+
+            harness.charm.remove_secret("unit", "operator-password")
+            assert error_message in caplog.text
 
-            self.harness.charm.remove_secret("app", "non-existing-secret")
+            harness.charm.remove_secret("app", "non-existing-secret")
             assert (
                 "Non-existing field 'non-existing-secret' was attempted to be removed"
-                in self._caplog.text
+                in caplog.text
             )
 
-            self.harness.charm.remove_secret("unit", "non-existing-secret")
+            harness.charm.remove_secret("unit", "non-existing-secret")
             assert (
                 "Non-existing field 'non-existing-secret' was attempted to be removed"
-                in self._caplog.text
+                in caplog.text
             )
 
-    @parameterized.expand([("app", True), ("unit", True), ("unit", False)])
-    @patch_network_get(private_address="1.1.1.1")
-    @patch("charm.PostgresqlOperatorCharm._on_leader_elected")
-    @patch("charm.JujuVersion.has_secrets", new_callable=PropertyMock, return_value=True)
-    def test_migration_from_databag(self, scope, is_leader, _, __):
-        """Check if we're moving on to use secrets when live upgrade from databag to Secrets usage."""
+
+@pytest.mark.parametrize("scope,is_leader", [("app", True), ("unit", True), ("unit", False)])
+@patch_network_get(private_address="1.1.1.1")
+def test_migration_from_databag(harness, juju_has_secrets, scope, is_leader):
+    """Check if we're moving on to use secrets when live upgrade from databag to Secrets usage."""
+    with (
+        patch("charm.PostgresqlOperatorCharm._on_leader_elected"),
+    ):
+        # as this test checks for a migration from databag to secrets,
+        # there's no need for this test when secrets are not enabled.
+        if not juju_has_secrets:
```

Comment:
here and on the test below, same idea from https://github.com/canonical/postgresql-operator/pull/425